### ISSUE TITLE
为什么paddle 加载训练好的模型，每次做predict，结果不一样呢？

### ISSUE THREAD

**Issue by jeffzhengye** [PaddlePaddle/Paddle#32328]
特意设置只有一个输入，# model.eval() 这个开关都试过。模型中有droptou，但是predict 接口设计的时候肯定考虑了这个问题啊。神奇啊

`def eval_integration():
    best_model = 'best_model.pdopt'
    best_model_paras = 'best_model.pdparams'
    if use_erine:
        model = ErnieForSequenceClassificationQ4.from_pretrained(MODEL_NAME_ERINE)
    else:
        model = BertForSequenceClassificationQ4.from_pretrained(MODEL_NAME)
    test_set = MyDataset(test_df.iloc[:1])
    test_set = MapDataset(test_set)
    test_set.map(train_trans_func, batched=False)

    # test_batch_sampler = paddle.io.BatchSampler(
    #     test_set, batch_size=batch_size, shuffle=False)

    test_data_loader = paddle.io.DataLoader(
        dataset=test_set,
        collate_fn=batchify_fn,
        shuffle=False,
        batch_size=64,
        return_list=True)

    predictions = []
    for fold in range(CFG['fold_num']):
        params_dict = paddle.load(f'./{model_outputs}{fold}/' + best_model)
        opt_dict = paddle.load(f'./{model_outputs}{fold}/' + best_model_paras)
        model.set_state_dict(params_dict)
        # model.eval()
        pmodel = paddle.Model(model)
        optimizer = paddle.optimizer.AdamW(
            learning_rate=learning_rate,
            parameters=model.parameters(),
            weight_decay=weight_decay)
        optimizer.set_state_dict(opt_dict)

        # Defines loss and metric.
        criterion = paddle.nn.CrossEntropyLoss()
        metric = paddle.metric.Accuracy()
        pmodel.prepare(optimizer, criterion, metric)
        y_pred = pmodel.predict(test_data_loader)
        y_pred = np.concatenate(y_pred[0])
        print('y_pred_shape', y_pred.shape)
        print(y_pred[0])
        predictions.append(y_pred)
        break`

**Comment by jeffzhengye** [comment]
> 您好，我们已经收到了您的问题，会安排技术人员尽快解答您的问题，请耐心等待。请您再次检查是否提供了清晰的问题描述、复现代码、环境&版本、报错信息等。同时，您也可以通过查看[官网API文档](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/index_cn.html)、[常见问题](https://www.paddlepaddle.org.cn/documentation/docs/zh/faq/index_cn.html)、[历史Issue](https://github.com/PaddlePaddle/Paddle/issues)、[AI社区](https://ai.baidu.com/forum/topic/list/168)来寻求解答。祝您生活愉快～
> 
> Hi! We've received your issue and please be patient to get responded. We will arrange technicians to answer your questions as soon as possible. Please make sure that you have posted enough message to demo your request. You may also check out the [API](https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/index_cn.html)，[FAQ](https://www.paddlepaddle.org.cn/documentation/docs/zh/faq/index_cn.html)，[Github Issue](https://github.com/PaddlePaddle/Paddle/issues) and [AI community](https://ai.baidu.com/forum/topic/list/168) to get the answer.Have a nice day!

在线等，谢谢。

**Comment by jeffzhengye** [comment]
问题点找到了，但是真心不明白为什么会出现这样的问题。

save的模型中参数，跟重新初始化模型中参数名全部都不一样。

以下是载入模型报的warning：

`/home/user/anaconda3/lib/python3.7/site-packages/paddle/fluid/dygraph/layers.py:1303:` UserWarning: Skip loading for ernie.pooler.dense.bias. ernie.pooler.dense.bias is
not found in the provided dict.`

模型save的代码如下：
`
        if use_erine:
            model = ErnieForSequenceClassificationQ4.from_pretrained(MODEL_NAME_ERINE)
        else:
            model = BertForSequenceClassificationQ4.from_pretrained(MODEL_NAME)
        # model = BertForSequenceClassificationQ4(bert=pretrained)
        pmodel = paddle.Model(model)
        # optimizer = paddle.optimizer.Adam(parameters=pmodel.parameters(), learning_rate=learning_rate)
        # lrscheduler = paddle.optimizer.lr.LRScheduler(learning_rate=learning_rate)

        num_training_steps = len(train_data_loader) * epochs
        lr_scheduler = ppnlp.transformers.CosineDecayWithWarmup(learning_rate, num_training_steps, warmup_proportion)
        optimizer = paddle.optimizer.AdamW(
            learning_rate=lr_scheduler,
            parameters=model.parameters(),
            weight_decay=weight_decay)

        # Defines loss and metric.
        criterion = paddle.nn.CrossEntropyLoss()
        metric = paddle.metric.Accuracy()
        pmodel.prepare(optimizer, criterion, metric)

        earlystop = paddle.callbacks.EarlyStopping('acc', mode='max', patience=4, verbose=1, save_best_model=True)
        # paddle.callbacks.LRScheduler
        # Starts training and evaluating.
        pmodel.fit(train_data_loader, eval_data=val_data_loader, epochs=epochs, verbose=1,
                   save_dir=f'./{model_outputs}{fold}',
                   num_workers=10,
                   callbacks=[earlystop])

`
